File: scripts/check-no-test-any.mjs
import { readdir, readFile } from "node:fs/promises";
import { join } from "node:path";

const roots = ["tests"];
const anyPattern = /\bany\b/;
const testFilePattern = /\.ts$/;
const lineBreakPattern = /\r?\n/;
const matches = [];

async function walk(dir) {
	for (const entry of await readdir(dir, { withFileTypes: true })) {
		const path = join(dir, entry.name);
		if (entry.isDirectory()) {
			await walk(path);
			continue;
		}
		if (!testFilePattern.test(path)) continue;
		const lines = (await readFile(path, "utf8")).split(lineBreakPattern);
		for (let i = 0; i < lines.length; i++) {
			if (anyPattern.test(lines[i]))
				matches.push(`${path}:${i + 1}: ${lines[i]}`);
		}
	}
}

for (const root of roots) await walk(root);

if (matches.length > 0) {
	console.error(matches.join("\n"));
	process.exit(1);
}

File: scripts/check_no_test_any.py
"""Reject uses of the bare ``any`` type in TypeScript tests."""

from __future__ import annotations

import re
import sys
from pathlib import Path

from . import iter_files

_ANY_PATTERN = re.compile(r"\bany\b", re.ASCII)
_LINE_BREAK_PATTERN = re.compile(r"\r?\n")


def main(
    argv: list[str] | None = None,
    repo_root: str | Path | None = None,
) -> int:
    """Run the test source ``any`` check."""
    del argv
    matches: list[str] = []
    for display_path, file_path in iter_files("tests", ".ts", repo_root):
        lines = _LINE_BREAK_PATTERN.split(file_path.read_text(encoding="utf-8"))
        for line_number, line in enumerate(lines, start=1):
            if _ANY_PATTERN.search(line):
                matches.append(f"{display_path}:{line_number}: {line}")

    if matches:
        print("\n".join(matches), file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
